lib/Listeners/NodeCopiedListener.php

```diff
@@ -36,6 +37,9 @@ public function handle(Event $event): void {
 			&& $target->getMimeType() === 'text/markdown'
 		) {
 			$this->attachmentService->copyAttachments($source, $target);
+			$target->unlock(ILockingProvider::LOCK_SHARED);
+			AttachmentService::replaceAttachmentFolderId($source, $target);
+			$target->lock(ILockingProvider::LOCK_SHARED);
 		}
```

Comment:
@mejo- This looks odd to me. First you unlock and then you lock. Am I missing something?

Reply:
Also... maybe you can push the locking down into the `AttachmentService`? Seems like something one would always want when running `replaceAttachmentFolderId`.

Reply:
Well, the problem is that there's still a shared lock on the target file, probably from the copy process. I didn't dig into it further yet, but since I only remove the lock in order to allow updating the content and don't want to interfere with the copy process otherwise, I decided to bring back the lock afterwards. That's also why I did this in the `NodeCopiedListener` directly, as it's only needed when `replaceAttachmentFolderId()` is called from inside `NodeCopiedEvent`.

Reply:
ah.. that makes sense.